### ISSUE TITLE
Remove outdated assumptions (all APIs support feature data and only feature data) 

### ISSUE THREAD

**Issue by cportele** [ldproxy/ldproxy#473]
Historically, each ldproxy API was an API that shared a feature dataset and each collection represented a feature type. This is no longer the case, but this assumption is still in many parts reflected in the design, the code and the terms used. The maintainability of the code would benefit from updating the internal APIs to reflect that ldproxy now supports all kinds of geospatial APIs and resource types.

**Comment by azahnen** [comment]
@cportele I think you improved this in the meantime, can this be closed?

**Comment by cportele** [comment]
Yes, I think this can be closed. If there is more, we should create new issues.
